Formula/oclgrind.rb

```diff
@@ -0,0 +1,18 @@
+class Oclgrind < Formula
+  desc "OpenCL device simulator and debugger"
+  homepage "https://github.com/jrprice/Oclgrind"
+  url "https://github.com/jrprice/Oclgrind/archive/v18.3.tar.gz"
+  sha256 "90518d47573e64c6c28e173dc6f10c4e0ca53a99543ef2f1afaac1cbf725fe90"
+
+  depends_on "cmake" => :build
+  depends_on "llvm" => :build
+
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/oclgrind", "ls", "-l"
```

Comment:
Could we match the output of this command? Or run a test that exercises the functionality of oclgrind a bit more?

Reply:
I can add a program that actually uses OpenCL, but it will be dozens of lines of crud (the OpenCL API is very verbose).  Most of the formula will just be a big C program.  Is this interesting?  Alternatively, is there a program preinstalled on macOS that calls OpenCL?

While the current test does not test much of oclgrind's functionality, it does test that it can set up its internal structures and load the LLVM stuff it needs.  In my experience, that is where oclgrind installations usually break.

Reply:
Running `make test` would be recommended, there's around 100 tests in there that mostly cover internal instruction handling, but also touch the OpenCL API implementation.

Reply:
Alright, I'll take a swing at that.

Reply:
In the `test do` block, we don't have access to the build files anymore, so `make check` will probably not be possible.

@athas can you look at the Open-MPI formula, as an example of the kind of tests we like?

Reply:
I can do something like that, but note that the smallest OpenCL program is significantly larger than those.  This is the smallest I can find: http://phycomp.technion.ac.il/~comphy/classfiles/hellocl.c (it will shrink a bit from removing comments and such, though).  Is that small enough?

Reply:
Yes, that will work, trimmed down by removing comments. Or maybe this one, which is even a bit shorted: https://us.fixstars.com/opencl/book/OpenCLProgrammingBook/first-opencl-program/